lib/src/code_generator/typealias.dart

```diff
@@ -122,3 +135,30 @@ class Typealias extends BindingType {
   String? getDefaultValue(Writer w, String nativeLib) =>
       type.getDefaultValue(w, nativeLib);
 }
+
+/// Objective C's instancetype.
+///
+/// This is an alias for an NSObject* that is special cased in code generation.
+class ObjCInstanceType extends Typealias {
```

Comment:
Would it be possible to populate the enclosing class here? So that the code generator doesn't make the decision but we do it here?